### PR TITLE
upgrade(ghc): changes necessary for ghc 8.10 and 9.0 compatibility

### LINKED ISSUE #47: docopt 0.7.0.5 fails to build with GHC 8.10.7 and 9.0.1
docopt 0.7.0.5 needs to be revised on hackage to `base < 4.14` since it does not build with GHC 8.10 or 9.0.  (I suspect versions 0.7.0.<5 have the same problem.)

Revisions can be made at e.g. https://hackage.haskell.org/package/docopt/maintain

Error (GHC 9.0.1):
```
[ 6 of 10] Compiling System.Console.Docopt.Public ( System/Console/Docopt/Public.hs, dist/build/System/Console/Docopt/Public.o, dist/build/System/Console/Docopt/Public.dyn_o )

System/Console/Docopt/Public.hs:195:17: error:
    • Could not deduce (MonadFail m) arising from a use of ‘fail’
      from the context: Monad m
        bound by the type signature for:
                   getFirstArg :: forall (m :: * -> *).
                                  Monad m =>
                                  Arguments -> Option -> m String
        at System/Console/Docopt/Public.hs:193:1-57
      Possible fix:
        add (MonadFail m) to the context of
          the inferred type of failure :: m a
          or the type signature for:
               getFirstArg :: forall (m :: * -> *).
                              Monad m =>
                              Arguments -> Option -> m String
```

See also https://matrix.hackage.haskell.org/#/package/docopt for where the bounds to not reflect buildability (look for the red FAIL entries).

### PR DESCRIPTION
**Pull request by simonmichael**
These make docopt build cleanly with current stackage/cabal/ghc-8.10/ghc-9.0.

Closes #47 #45

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@ cabal.sandbox.config
 # extra
 tmp
 .stack-work
+stack.yaml.lock
```
Comment (FullstackGJJ) [inline .gitignore:24]: How come you wanted to remove the stack.yaml.lock? I thought it was generally recommended to check that into source control https://docs.haskellstack.org/en/stable/lock_files/#lock-files

File: stack.yaml.lock [hunk 2]
```diff
@@ -1,12 +0,0 @@
-# This file was autogenerated by Stack.
```
Comment (FullstackGJJ) [inline stack.yaml.lock:1]: Same question asked above. I thought it was generally recommended to check that into source control https://docs.haskellstack.org/en/stable/lock_files/#lock-files

**Review by FullstackGJJ — CHANGES_REQUESTED** [review]
Want to address the removal of the stack.yaml.lock

File: stack.yaml.lock (hunk 2 above)
Comment (simonmichael) [reply stack.yaml.lock:1]: Thanks for looking at this @FullstackGJJ. To answer your question, I have been using stack for a long time and have never seen a need to commit that lock file. I know that the idea is to increase reproducility somehow, but I believe it's just not necessary in normal FOSS projects. And when it is committed, I find it always has changes that get in the way of my work. (I don't know the precise rules for when it changes, and don't really want to..)

Comment (FullstackGJJ) [reply stack.yaml.lock:1]: Sounds good. Resolving these comments.

**Pull request merged**